persistence/sql/sql_process_store_common.go

```diff
@@ -186,58 +194,76 @@ func (p sqlProcessStoreImpl) getDedupIdToLocalQueueMessageMap(ctx context.Contex
 }
 
 func (p sqlProcessStoreImpl) updateCommandResultsWithConsumedLocalQueueMessages(
-	commandResults *xdbapi.CommandResults,
-	consumedMessages []persistence.InternalLocalQueueMessage,
+	commandResults *persistence.CommandResultsJson,
+	consumedMessagesMap map[int][]persistence.InternalLocalQueueMessage,
 	dedupIdToLocalQueueMessageMap map[string]extensions.LocalQueueMessageRow) error {
 
-	for _, consumedMessage := range consumedMessages {
-		message, ok := dedupIdToLocalQueueMessageMap[consumedMessage.DedupId]
-		if !ok {
-			continue
-		}
+	if len(commandResults.LocalQueueResults) == 0 {
+		commandResults.LocalQueueResults = map[int][]xdbapi.LocalQueueMessageResult{}
```

Comment:
The commandResults was generated by `persistence.NewCommandResultsJson()`, but after getting it back from the table, we still need to initialize the maps?